dozer-ingestion/src/connectors/mysql/conversion.rs

```diff
@@ -190,3 +187,202 @@ impl GeomProcessor for PointProcessor {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::time::Duration;
+
+    use crate::connectors::mysql::conversion::{IntoField, IntoFields};
+
+    use super::get_field_type_for_mysql_column_type;
+    use chrono::{DateTime, NaiveDate, NaiveDateTime, NaiveTime, Offset, Utc};
+    use dozer_types::{
+        json_types::JsonValue,
+        types::{DozerDuration, Field, FieldType, TimeUnit},
+    };
+    use mysql_async::Column;
+    use mysql_common::{constants::ColumnType, row::new_row, Value};
+
+    #[test]
+    fn test_field_types() {
```

Comment:
I think it might be more useful to extract the parsing logic for data types into a separate function and test that (so, test that "decimal(5, 2)" parses to "decimal"), instead of basically replicating the match statement in that function in the test.

A metric I like to use for deciding if something is worth testing, is if the test would be of a lower order of complexity than the thing that's being tested. In this case, most of the test tests a simple match statement and therefore you are just as likely to make a mistake writing the test as you are writing the implementation.

Reply:
I don't think changing how it works would make it harder to understand (the structure of the test would be the same, just with slightly different data).

I agree it isn't worth changing it now, as you have already done the work, the values won't change, and it's definitely not a bad thing to be thorough!